### PR TITLE
Add support for group analytics into posthog-node

### PR DESCRIPTION
**Pull request by macobo**
Part of https://github.com/PostHog/posthog/issues/7050

Oops, actually posthog-js-lite is used there. :sweat_smile: Still a valid PR

Not sure why feature flags tests fail - seems unrelated, cc @yakkomajuri?

Integration-tested this myself as well.

## Checklist
- [x] Tests for new code (if applicable)
- [x] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

**Comment by yakkomajuri** [comment]
https://github.com/PostHog/posthog-node/pull/49 broke stuff

Fixed it in https://github.com/PostHog/posthog-node/pull/49 and merged master here

Would prefer to leave the review to @neilkakkar if ok - have a lot of stuff on my plate and would rather help where I might be the "best choice" e.g. Go, Ruby, PHP (not that I'm great at these, just have better context). 

Everyone in our team knows JS though :D

File: index.js [hunk 1]
```diff
@@ -170,6 +175,30 @@ class PostHog {
...
+     * @param groupType Type of group (ex: 'company'). Limited to 5 per project
+     * @param groupKey Unique identifier for that type of group (ex: 'id:5')
+     * @param properties OPTIONAL | which can be a object with any information you'd like to add
+     */
+    groupIdentify(message, callback) {
+        this._validate(message, 'groupIdentify')
+
+        const captureMessage = {
+            event: '$groupidentify',
+            distinctId: `\$${message.groupType}_${message.groupKey}`,
```
Comment (neilkakkar) [inline index.js:191]: Seems different from the docs: https://posthog.com/docs/user-guides/group-analytics#updating-group-properties (distinctId inside properties vs outside)

**Review by neilkakkar — APPROVED** [review]
code looks good to me! Haven't tried running it locally though.

File: index.js (hunk 1 above)
Comment (macobo) [reply index.js:191]: Good question. This is intentional - `.capture` expects distinctId at top level and then munges it into the right spot in the actual API message.

https://github.com/PostHog/posthog-node/blob/master/index.d.ts#L24-L33

**Pull request merged**